### DISCUSSION TITLE
Moving Nav Component out of the Months Component

### DISCUSSION THREAD

**Discussion by elliottcrif** [gpbl/react-day-picker#2651] [Ideas]
What do we think about moving the Nav component out of the Months section to allow more flexibility in layout. For example:

```tsx

<components.Nav
  className={classNames[UI.Nav]}
  style={styles?.[UI.Nav]}
  aria-label={labelNav()}
  onPreviousClick={handlePreviousClick}
  onNextClick={handleNextClick}
  previousMonth={previousMonth}
  nextMonth={nextMonth}
/>

<components.Months
  className={classNames[UI.Months]}
  style={styles?.[UI.Months]}
>
  {months.map((calendarMonth, displayIndex) => (
    <components.Month ... />
  ))}
</components.Months>
```

I am currently having some difficulty making the styling of the months render differently than the nav i.e months rendering in a flexbox and the nav at the top

**Comment by gpbl** [comment]
@elliottcrif, you’re absolutely right! Thanks for your suggestion!

I was planning to reorganize the navigation element to address the issue in https://github.com/gpbl/react-day-picker/issues/2629. However, this could potentially be a breaking change, so I was considering introducing a new prop to set a “navigation layout.”

Do you have some more feedback for improving the navigation?

**Reply by elliottcrif** [reply]
@gpbl I don't have much more feedback, that sounds like a great approach to me.
